tests/test_mean_estimator.py

```diff
@@ -1,733 +1,166 @@
-                    1.28441757e-01,
-                    -5.57623000e-01,
-                    -1.54801935e-01,
-                    6.51729903e-01,
-                    -2.15567768e-01,
-                    1.95041528e-01,
-                    -4.18334680e-01,
-                    3.26735913e-02,
-                    6.35474331e-02,
-                    3.06828631e-01,
-                    1.43149180e-01,
-                    -2.34377520e-01,
-                    -1.54299735e-01,
-                    2.82627560e-01,
-                    9.60630473e-02,
-                    1.47687304e-01,
-                    1.38157247e-01,
-                    -4.25581692e-01,
-                    -5.62236939e-01,
-                    2.09287213e-01,
-                    -1.14280315e-01,
-                    2.70617650e-02,
-                    -1.19705716e-01,
-                    1.68350236e-02,
-                    -1.20459065e-01,
-                    6.03971532e-02,
-                    3.21465643e-01,
-                    1.82032297e-01,
-                    -2.95991444e-02,
-                    1.53711400e-01,
-                    -1.30594319e-01,
-                    -4.71412485e-02,
-                    -1.35301477e-01,
-                    -2.36292616e-01,
-                    1.95728111e-01,
-                    2.54618329e-01,
-                    -1.81663289e-03,
-                    2.77960420e-02,
-                    3.58816749e-02,
-                    -2.50138365e-02,
-                    2.54103161e-01,
-                    9.82534014e-02,
-                    9.00807559e-02,
-                    3.71458771e-02,
-                    -7.86838200e-02,
-                    -1.03837231e-01,
-                    -1.26116949e-01,
-                    9.82006976e-02,
-                ],
-                atol=1e-4,
-            )
+@pytest.fixture(scope="module")
+def mask(L):
+    return grid_3d(L)["r"] < 1
+
+
+# Tests
+def test_resolution_error(sim, basis):
+    """
+    Test mismatched resolutions yields a relevant error message.
+    """
+
+    with raises(ValueError, match=r".*resolution.*"):
+        # This basis is intentionally the wrong resolution.
+        incorrect_basis = FBBasis3D(sim.L + 1, dtype=sim.dtype)
+
+        _ = MeanEstimator(sim, basis=incorrect_basis, preconditioner="none")
+
+
+def test_estimate(sim, estimator, mask):
+    estimate = estimator.estimate()
+
+    est = estimate * mask
+    vol = sim.vols * mask
+
+    np.testing.assert_allclose(
+        est / np.linalg.norm(est), vol / np.linalg.norm(vol), atol=0.1
+    )
+
+
+def test_adjoint(sim, basis, estimator, mask):
+    # Mean coefs formed by backprojections
+    mean_b_coef = estimator.src_backward()
+
+    # Evaluate mean coefs into a volume
+    est = Coef(basis, mean_b_coef).evaluate() * mask
+
+    # Mask off corners of volume
+    vol = sim.vols * mask
+
+    # Assert the mean volume is close to original volume
+    np.testing.assert_allclose(
```

Comment:
Why should this work? The `mean_b_coeff` contains the right-hand side coefficients of the normal equation. To get an estimate of the mean volume, you need to invert the kernel first.

Reply:
I thought this was an more/less an unfiltered backprojection and should be similar to the original volume.  Is that not the right way to think about it?

It does capture the effect of the bug.

Reply:
discussed in team meeting, replaced test.